### DISCUSSION TITLE
Checking For Form Field Updates (Dirty Values)

### DISCUSSION THREAD

**Discussion by discoverlance-com** [filamentphp/filament#11620] [Help]
### Package

Panel builder

### Package Version

v3.2.34

### How can we help you?

I want to be able to check if the form fields or values are dirty before saving in a create resource page from the admin panel.

Basically, when the user fills in the form the first time and submits, if it fails validation and I return some kind of an error, I want to be able to check the filled in values from the form to see if any of them have changed compared to the previous values. For instance, has the user changed the name since the last submission. The closest I could find when using livewire is [showing dirty indicators](https://livewire.laravel.com/docs/forms#showing-dirty-indicators) which is used in a live update form for instance to inform the user when the value they entered has not been saved just yet to the database.

Is there a way I can achieve the behaviour I described in filament?

**Comment by iancallahan** [comment]
This is how I did it using the handleRecordUpdate method on the Edit page class for the resource (in my case, needed to keep track of previously changed fields as well):


```
protected function handleRecordUpdate(Model $record, array $data): Model
 {
        $old_fields = $record->changed_fields;
        
        $record->fill($data);
        
        $changed_fields = array_merge($old_fields, $record->getDirty());
        
        $data['changed_fields'] = $changed_fields;

        $record->update($data);
    
        return $record;
}
```
